source/MaterialXTest/MaterialXCore/Document.cpp

```diff
@@ -116,3 +118,48 @@ TEST_CASE("Document", "[document]")
     // Validate the combined document.
     REQUIRE(doc->validate());
 }
+
+TEST_CASE("Document equivalence", "[document2]")
+{
+    mx::DocumentPtr doc = mx::createDocument();
+    std::multimap<std::string, std::string> inputMap;
+
+    inputMap.insert({ "color3", "  1.0,   +2.0,  3.0   " });
+    inputMap.insert({ "color4", "1.0,   2.00, 0.3000, -4" });
+    inputMap.insert({ "float", "  00.1000  " });
+    inputMap.insert({ "vector2", "1.0,   0.012345678" });
+    inputMap.insert( {"vector3", "  1.0,   +2.0,  3.0   " });
+    inputMap.insert( {"vector4", "1.0,   2.00, 0.3000, -4" });
+
+    unsigned int index = 0;
+    for (auto it = inputMap.begin(); it != inputMap.end(); ++it)
+    {
+        mx::InputPtr input= doc->addInput("input" + std::to_string(index), (*it).first);
+        input->setValueString((*it).second);
+        index++;
+    }
+
+    mx::DocumentPtr doc2 = mx::createDocument();
+    std::multimap<std::string, std::string> inputMap2;
+    inputMap2.insert({ "color3", "1,2,3" });
+    inputMap2.insert({ "color4", "1,2,0.3,-4" });
+    inputMap2.insert( {"float", "0.1" });
+    inputMap2.insert( {"vector2", "1,0.0123456" });
+    inputMap2.insert( {"vector3", "1,2,3" });
+    inputMap2.insert( {"vector4", "1,2,0.3,-4" });
+
+    index = 0;
+    for (auto it = inputMap.begin(); it != inputMap.end(); ++it)
+    {
+        mx::InputPtr input = doc2->addInput("input" + std::to_string(index), (*it).first);
+        input->setValueString((*it).second);
+        index++;
+    }
+
+    doc->normalizeValueStrings();
+    doc2->normalizeValueStrings();
+
+    // Note: do not check doc2 == doc as that is a pointer comparison
```

Comment:
Not sure if there is a way to allow pointer comparison to be a value comparison, but left a note here since I ran into it.